evm/logic/call.py

```diff
@@ -58,8 +58,9 @@ def __call__(self, computation):
         computation.gas_meter.consume_gas(child_msg_gas_fee, reason=self.mnemonic)
 
         # Pre-call checks
-        state_db = computation.vm_state.read_only_state_db
-        sender_balance = state_db.get_balance(computation.msg.storage_address)
+        with computation.state_db(read_only=True) as state_db:
+            sender_balance = state_db.get_balance(computation.msg.storage_address)
+
```

Comment:
TODO: After this is merged, I will make a PR to remove `Message.access_list` and `Computation.state_db` which should only apply to `ShardingVM` and another PR to convert this `state_db(read_only=True)`/`state_db()` back to `read_only_state_db`/`mutable_state_db`.